x-pack/plugins/uptime/server/rest_api/synthetics_service/add_monitor.ts

```diff
@@ -54,10 +54,12 @@ export const addSyntheticsMonitorRoute: UMRestApiRouteFactory = () => ({
       formatTelemetryEvent({ monitor: newMonitor, errors, kibanaVersion: server.kibanaVersion })
     );
 
-    if (errors) {
-      return errors;
+    if (errors && errors.length > 0) {
+      return response.badRequest({
```

Comment:
```suggestion
      return response.ok({
```
We should return a response.ok in this scenario. The errors returned from the service are treated as warnings, the monitor is still saved. We should treat the main action from this API as saving the monitor, and syncing to the service as a side effect. 

Currently, the data is not being sent back via `useFetcher` when a non 200 status is returned. This results in a generic error toast. 
<img width="457" alt="Screen Shot 2022-03-08 at 2 12 50 PM" src="https://user-images.githubusercontent.com/11356435/157309528-8b421871-2c99-481a-99b8-fbc8022b4305.png">

It should instead be displaying the warning message.
<img width="424" alt="Screen Shot 2022-03-08 at 2 22 04 PM" src="https://user-images.githubusercontent.com/11356435/157309968-a9ee722a-201c-4dd2-96a9-186df943b70c.png">

Reply:
Thanks for noticing this! I've just applied the patch and tested by:

* Not having certs and having incorrect credentials
* Trying to create a monitor
* Ensuring the warning appears, not the error